test/XRP/serializer.test.ts

```diff
@@ -1937,4 +1939,95 @@ describe('serializer', function (): void {
     // THEN the result is as expected.
     assert.equal(serialized, Utils.toHex(fulfillmentBytes))
   })
+
+  it('Serializes an EscrowFinish with required fields', function (): void {
+    // GIVEN an EscrowFinish with required fields.
+    const offerSequence = new OfferSequence()
+    offerSequence.setValue(offerSequenceNumber)
+
+    const owner = new Owner()
+    owner.setValue(testAccountAddress)
+
+    const escrowFinish = new EscrowFinish()
+    escrowFinish.setOfferSequence(offerSequence)
+    escrowFinish.setOwner(owner)
+
+    // WHEN it is serialized.
+    const serialized = Serializer.escrowFinishToJSON(escrowFinish)
+
+    // THEN the result is as expected.
+    const expected: EscrowFinishJSON = {
+      OfferSequence: Serializer.offerSequenceToJSON(offerSequence),
+      Owner: Serializer.ownerToJSON(owner)!,
+      TransactionType: 'EscrowFinish',
+    }
+
+    assert.deepEqual(serialized, expected)
+  })
+
+  it('Serializes an EscrowFinish with all fields', function (): void {
+    // GIVEN an EscrowFinish with all fields.
+    const conditionBytes = new Uint8Array([0, 1, 2, 3])
+    const condition = new Condition()
+    condition.setValue(conditionBytes)
+
+    const fulfillmentBytes = new Uint8Array([0, 1, 2, 3])
+    const fulfillment = new Fulfillment()
+    fulfillment.setValue(fulfillmentBytes)
+
+    const offerSequence = new OfferSequence()
+    offerSequence.setValue(offerSequenceNumber)
+
+    const owner = new Owner()
+    owner.setValue(testAccountAddress)
+
+    const escrowFinish = new EscrowFinish()
+    escrowFinish.setCondition(condition)
+    escrowFinish.setFulfillment(fulfillment)
+    escrowFinish.setOfferSequence(offerSequence)
+    escrowFinish.setOwner(owner)
+
+    // WHEN it is serialized.
+    const serialized = Serializer.escrowFinishToJSON(escrowFinish)
+
+    // THEN the result is as expected.
+    const expected: EscrowFinishJSON = {
+      Condition: Serializer.conditionToJSON(condition),
+      Fulfillment: Serializer.fulfillmentToJSON(fulfillment),
+      OfferSequence: Serializer.offerSequenceToJSON(offerSequence),
+      Owner: Serializer.ownerToJSON(owner)!,
+      TransactionType: 'EscrowFinish',
+    }
+
+    assert.deepEqual(serialized, expected)
+  })
+
+  it('Fails to serialize an EscrowFinish missing required fields', function (): void {
+    // GIVEN an EscrowFinish that's missing required fields.
+    const escrowFinish = new EscrowFinish()
+
+    // WHEN it is serialized.
+    const serialized = Serializer.escrowFinishToJSON(escrowFinish)
+
+    // THEN the result is undefined.
+    assert.isUndefined(serialized)
+  })
+
+  it('Fails to serialize an EscrowFinish with a malformed owner', function (): void {
+    // GIVEN an EscrowCancel with a malformed owner.
+    const offerSequence = new OfferSequence()
+    offerSequence.setValue(offerSequenceNumber)
+
+    const owner = new Owner()
+
+    const escrowFinish = new EscrowFinish()
+    escrowFinish.setOfferSequence()
```

Comment:
you need to pass `offerSequence` here to make this test work correctly.

Reply:
for sure, will do it